pkg/routeagent/controllers/route/driver.go

```diff
@@ -3,32 +3,59 @@ package route
 import (
 	"fmt"
 	"io/ioutil"
+	"net"
 
-	"github.com/vishvananda/netlink"
 	"k8s.io/klog"
 )
 
-const FlannelInterface = "flannel.1"
+func discoverCNIInterface(clusterCIDR string) *cniInterface {
+	_, clusterNetwork, err := net.ParseCIDR(clusterCIDR)
+	if err != nil {
+		klog.Errorf("Unable to ParseCIDR %q : %v", clusterCIDR, err)
+		return nil
+	}
 
-func isFlannelCNI() error {
-	_, err := netlink.LinkByName(FlannelInterface)
+	hostInterfaces, err := net.Interfaces()
 	if err != nil {
-		return fmt.Errorf("error retrieving link by name %s: %v", FlannelInterface, err)
+		klog.Errorf("net.Interfaces() returned error : %v", err)
+		return nil
 	}
 
-	// FlannelInterface exists, return nil
+	for _, iface := range hostInterfaces {
+		addrs, err := iface.Addrs()
+		if err != nil {
+			klog.Errorf("For interface %q, iface.Addrs returned error: %v", iface.Name, err)
+			return nil
+		}
+
+		if len(addrs) > 0 {
```

Comment:
nit - this check isn't needed as `range` covers it on the next line.

Reply:
Done